packages/aws-lambda-otel-extension/opt/otel-extension/external/otel-payloads.js

```diff
@@ -118,6 +118,76 @@ const createResourceAttributes = (fun) =>
     }
   });
 
+const createLogPayload = (fun, logs) => {
+  const spanData = fun.span;
+  const key = Object.keys(fun.eventData)[0];
+  const metricsAtt = createMetricAttributes({ record: fun.eventData[key] }, {})
+    .filter((attr) => {
+      return [
+        'faas.arch',
+        'faas.api_gateway_request_id',
+        'faas.event_source',
+        'faas.api_gateway_app_id',
+      ].includes(attr.key);
+    })
+    .reduce(
+      (obj, attr) => ({
+        ...obj,
+        [attr.key]: Object.values(attr.value)[0],
+      }),
+      {}
+    );
```

Comment:
Wouldn't be more readable as:

```javascript
const metricAttributeNames = new Set(['faas.arch', ....])
const metricAttributes = {};
for (const attribute of createMetricAttributes({ record: fun.eventData[key] }, {})) {
  if (!metricAtributeNames.has(attr.key)) continue;
  metricsAttributes[attr.key] = atr.value;
}
```

I think `reduce` serves a bit different use cases, and with logic as proposed currently, it takes a while to seclude intention behind the logic.

What do you think?

Reply:
Dan, did you see that one?

Reply:
Ahh sorry I missed this one I can get this updated 👍

Reply:
Alright this should be updated to your suggestion 👍